projects/core/src/lib/users/user.service.ts

```diff
@@ -190,49 +195,55 @@ export class UserService {
      */
     login(credentials: {email: string; password: string}): Observable<Session> {
         const result = new ReplaySubject<Session>();
-        this.backend
-            .loginUser(credentials)
-            .pipe(mergeMap((response) => this.createSession(response)))
-            .subscribe(
-                (session) => {
-                    this.session$.next(session);
-                    result.next(session);
+
+        new SessionApi(new Configuration({basePath: this.config.API_URL}))
+            .loginHandler({
+                userCredentials: {
+                    email: credentials.email,
+                    password: credentials.password,
                 },
-                (error) => result.error(error),
-                () => result.complete(),
-            );
+            })
+            .then((response) => this.sessionFromDict(response))
+            .then((session) => {
+                this.session$.next(session);
+                result.next(session);
+                result.complete();
+            })
+            .catch((error) => result.error(error));
+
         return result.asObservable();
     }
 
     guestLogin(): Observable<Session> {
         const result = new ReplaySubject<Session>();
         this.session$.pipe(first()).subscribe((oldSession) => {
             if (oldSession) {
-                this.backend.logoutUser(oldSession.sessionToken).subscribe();
+                new SessionApi(new Configuration({basePath: this.config.API_URL, accessToken: oldSession.sessionToken})).logoutHandler();
             }
 
-            this.createGuestUser().subscribe(
-                (session) => {
+            this.createGuestUser().subscribe({
+                next: (session) => {
                     this.session$.next(session);
                     result.next(session);
                 },
-                (error) => {
+                error: (error) => {
                     // failing on a guest login means we cannot do it,
                     // so we are logged out
                     this.session$.next(undefined);
                     result.error(error);
                 },
-                () => result.complete(),
-            );
+                complete: () => result.complete(),
+            });
         });
         return result.asObservable();
     }
 
     createSessionWithToken(sessionToken: UUID): Observable<Session> {
-        return this.backend.getSession(sessionToken).pipe(
-            mergeMap((response) => this.createSession(response)),
-            tap((session) => this.session$.next(session)),
-        );
+        return from(
+            new SessionApi(new Configuration({basePath: this.config.API_URL, accessToken: sessionToken}))
+                .sessionHandler()
```

Comment:
I find these names rather confusing. I think it would be better if it's called getSessionHandler. I know it's the same in Python, so maybe we should think about renaming the handlers in Rust?

Reply:
Ok, we should discuss it, but not in this task.